sqlx-core/src/connection.rs

```diff
@@ -166,7 +166,27 @@ pub trait ConnectOptions: 'static + Send + Sync + FromStr<Err = Error> + Debug +
     where
         Self::Connection: Sized;
 
-    /// Log executed statements with the specified `level`
+    /// Log executed statements with the specified `level` using [LevelFilter]
+    ///
+    /// # Example (with Postgres)
+    ///
+    /// ```rust
+    /// use sqlx::{Pool, Postgres, postgres::PgConnectOptions, ConnectOptions};
+    ///
+    /// // First, you need to create ConnectOptions
+    /// let mut options = PgConnectOptions::new()
+    ///     .host("localhost").port("5432")
+    ///     .username("myuser").password("somesecretpassword")
+    ///     .database("mydb");
+    ///
+    /// // Now it's possible to change the log level of log statements
+    /// options.log_statements(LevelFilter::Trace);
+    ///
+    /// // Finally, connect to the DB using a single connection...
+    /// let conn = options.connect().await.unwrap();
+    /// // ... or a pool
+    /// let pool = Pool::<Postgres>::connect_with(options).await.unwrap();
+    /// ```
```

Comment:
You need to wrap this code in an `async fn` so the doctest compiles. See this snippet for an example: https://github.com/launchbadge/sqlx/blob/9ca1fbf2ca94a2bd653051350f9049b251a037cb/sqlx-core/src/postgres/options/mod.rs#L55-L77

Reply:
Actually that one isn't my favorite. I've preferred to do something like this:

````rust
/// ```
/// # async fn __whatever() -> sqlx::Result<()> {
/// // Actual body of example
///
/// # Ok(())
/// # }
/// ```
````

Reply:
@holubond did you see this?